### ISSUE TITLE
Support object spreads in function parameter

### ISSUE THREAD

**Issue by gre** [eslint/eslint#4102]
```js
const Foo = ({ foo, ...rest }) => rest;
```

results of

*Parsing error: Invalid left-hand side in formals list*

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
As indicated, this is not an error with ESLint, but rather a parser error with Espree. The issue should be filed at https://github.com/eslint/espree/issues/new.

**Comment by colinmeinke** [comment]
@gre did this get resolved?
